packages/elements/src/component-factory-strategy.ts

```diff
@@ -71,78 +71,108 @@ export class ComponentNgElementStrategy implements NgElementStrategy {
    */
   private readonly unchangedInputs = new Set<string>();
 
-  constructor(private componentFactory: ComponentFactory<any>, private injector: Injector) {}
+  /** check for zone.js loaded */
+  private readonly zoneless: boolean;
+
+  /** service for setting zone context */
+  private readonly ngZone: NgZone;
+
+  /** the zone the element was created in */
+  private readonly elementZone: Zone|undefined;
```

Comment:
Can we move the initialization here?

```suggestion
  private readonly elementZone = (typeof Zone === 'undefined') ? null : this.ngZone.run(() => Zone.current);
```